### PR TITLE
Add support for Jigoshop affiliate coupons

### LINKED ISSUE #229: Jigoshop coupon tracking
Jigoshop needs coupon tracking.

Related #93

### PR DESCRIPTION
**Pull request by ghost1227**
Fixes #229

### CODE REVIEW

File: includes/integrations/class-jigoshop.php [hunk 1]
```diff
@@ -42,8 +45,10 @@ public function init() {
 	 * @access public
 	 */
 	public function add_pending_referral( $order_id = 0 ) {
+		// Check if an affiliate coupon was used
+		$affiliate_id = $this->get_coupon_affiliate_id();
 
-		if( $this->was_referred() ) {
+		if( $this->was_referred() || $affiliate_id ) {
```
Comment (pippinsplugins) [inline includes/integrations/class-jigoshop.php]: Set `$this->affiliate_id = $affiliate_id` if an ID was pulled from the coupon.

File: includes/integrations/class-jigoshop.php [hunk 2]
```diff
@@ -135,6 +140,105 @@ public function reference_link( $reference = 0, $referral ) {
...
+	}
+
+	/**
+	 * Retrieve the affiliate ID for the coupon used, if any
+	 *
+	 * @access  public
+	 * @since   1.7.5
+	*/
+	private function get_coupon_affiliate_id() {
+		$coupons = jigoshop_cart::get_coupons();
```
Comment (pippinsplugins) [inline includes/integrations/class-jigoshop.php]: What about sites with hundreds or thousands of coupons?

File: includes/integrations/class-jigoshop.php (hunk 1 above)
Comment (ghost1227) [inline includes/integrations/class-jigoshop.php]: Oops... Thought I did that.

File: includes/integrations/class-jigoshop.php (hunk 2 above)
Comment (ghost1227) [inline includes/integrations/class-jigoshop.php]: What about them? :confused:

Comment (pippinsplugins) [inline includes/integrations/class-jigoshop.php]: Oh wait, this just pulls coupons from the cart. Sorry, I read this as though it was pulling all coupons from the site.

Comment (ghost1227) [inline includes/integrations/class-jigoshop.php]: Oh! I think you're misunderstanding... jigoshop_cart::get_coupons() gets all coupons _currently applied to the cart_

**Comment by pippinsplugins** [comment]
Made a new PR against 1.8: https://github.com/AffiliateWP/AffiliateWP/pull/1203

**Pull request closed without merging**